cmd/root.go

```diff
@@ -49,73 +60,49 @@ triage.`,
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
 	if err := rootCmd.Execute(); err != nil {
-		fmt.Println(err)
+		log.Println(err)
 		os.Exit(1)
 	}
 }
 
-// This function is useful for understanding what a particular viper contains.
-// It is more a crutch for development than anything I would ever expect a customer to see.
-func viperWiper(v *viper.Viper) {
-	fmt.Print("\n === Viper Wiper === \n\n")
-	for _, name := range v.AllKeys() {
-		fmt.Println("Key: ", name, " => Name:", v.GetString(name))
+func initConfig() {
+	if cfgFile != "" {
+		// log.Println("Using config file from the flag")
+	} else {
+		// log.Println("Using default config location")
+		cfgFile = "system_config.yaml"
 	}
-	fmt.Print("\n === Viper Wiper Done === \n\n")
-}
 
-// This function maps all pflags to strings in viper
-func initializeFlagswithViper(cmd *cobra.Command) error {
-	v := viper.GetViper()
+	cfgFile, _ := filepath.Abs(cfgFile)
 
-	if cfgFile != "" {
-		// Use config file from the flag.
-		v.SetConfigFile(cfgFile)
+	if _, ok := os.Stat(cfgFile); os.IsNotExist(ok) {
+		// It's okay if there isn't a config file
+		// log.Println("OK not using a config")
 	} else {
-		// Find home directory.
-		home, err := homedir.Dir()
-		if err != nil {
-			log.Fatalf("Home Directory not found: %v", err)
-		}
-		// Add the home directory to the config path
-		v.AddConfigPath(home)
+		// log.Println("Config file exists:", cfgFile)
 
-		// Add the local directory to the config path
-		v.AddConfigPath(".")
-		// Set the base name of the config file, without the file extension.
-		v.SetConfigName(defaultConfigFilename)
-	}
+		// Set the config file from the logic and checks above
+		viper.SetConfigFile(cfgFile)
 
-	// Attempt to read the config file, gracefully ignoring errors
-	// caused by a config file not being found. Return an error
-	// if we cannot parse the config file.
-	if err := v.ReadInConfig(); err != nil {
-		// It's okay if there isn't a config file
-		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
-			return err
+		if err := viper.ReadInConfig(); err == nil {
+			// log.Println("Reading in config:", viper.ConfigFileUsed())
```

Comment:
Same here.

Reply:
Fixed with refactor of loading this config file